Mathlib/Algebra/Category/ModuleCat/Presheaf/Sheafification.lean

```diff
@@ -0,0 +1,119 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.Algebra.Category.ModuleCat.Presheaf.Sheafify
+import Mathlib.CategoryTheory.Sites.LocallyBijective
+
+/-!
+# The sheafification functor for presheaves of modules
+
+In this file, we construct a functor
+`PresheafOfModules.sheafification α : PresheafOfModules R₀ ⥤ SheafOfModules R`
+for a locally bijective morphism `α : R₀ ⟶ R.val` where `R₀` is a presheaf of rings
+and `R` a sheaf of rings.
+In particular, if `α` is the identity of `R.val`, we obtain the
+sheafification functor `PresheafOfModules R.val ⥤ SheafOfModules R`.
+
+-/
+
+universe v v' u u'
+
+open CategoryTheory Category
+
+variable {C : Type u'} [Category.{v'} C] {J : GrothendieckTopology C}
+  {R₀ : Cᵒᵖ ⥤ RingCat.{u}} {R : Sheaf J RingCat.{u}} (α : R₀ ⟶ R.val)
+  [Presheaf.IsLocallyInjective J α] [Presheaf.IsLocallySurjective J α]
+  [HasWeakSheafify J AddCommGroupCat.{v}]
+  [J.WEqualsLocallyBijective AddCommGroupCat.{v}]
+
+namespace PresheafOfModules
+
+/-- Given a locally bijective morphism `α : R₀ ⟶ R.val` where `R₀` is a presheaf of rings
+and `R` a sheaf of rings (i.e. `R` identifies to the sheafification of `R₀`), this is
+the associated sheaf of modules functor `PresheafOfModules.{v} R₀ ⥤ SheafOfModules.{v} R`. -/
+@[simps map]
+noncomputable def sheafification : PresheafOfModules.{v} R₀ ⥤ SheafOfModules.{v} R where
+  obj M₀ := sheafify α (CategoryTheory.toSheafify J M₀.presheaf)
+  map f := sheafifyMap _ _ _ f ((presheafToSheaf J AddCommGroupCat).map f.hom) (by simp)
+  map_id M₀ := by
+    ext1
+    apply (toPresheaf _).map_injective
+    simp [toPresheaf, sheafify]
+  map_comp _ _ := by
+    ext1
+    apply (toPresheaf _).map_injective
+    simp [toPresheaf, sheafify]
+
+attribute [-simp] sheafification_map
```

Comment:
Is this local or global? If latter, you can use `simps (config := .lemmasOnly)` in the `simps` above.

Reply:
Thanks for the suggested syntax!